### ISSUE TITLE
Idea to Factor Out Common Action Logic

### ISSUE THREAD

**Issue by jbw3** [evanofslack/pyminion#110]
I had an idea to factor out the common play logic for action classes.
Currently, every derived `Action` class has to copy/paste the following at the beginning of its `play()` function:
```py
logger.info(f"{player} plays {self}")

if generic_play:
    super().generic_play(player)
```

I propose factoring this common logic out into the base `Action` `play()` method and calling the base function in derived classes:
```py
super().play(player, game, generic_play)
```

It's possible we could take this a step further and have the base `play()` method handle the basic +cards, +actions, +money, and +buys:
```py
def play(self, player: "Player", game: "Game", generic_play: bool = True) -> None:
    logger.info(f"{player} plays {self}")

    if generic_play:
        super().generic_play(player)

    if self.draw > 0:
        player.draw(self.draw)
    player.state.actions += self.actions
    player.state.money += self.money
    player.state.buys += self.buys
```
As a result, vanilla cards like Village, Market, etc. would not need to implement a `play()` method at all.

What are your thoughts?

**Comment by evanofslack** [comment]
Yup this would be a great addition

**Comment by evanofslack** [comment]
closed by #113
